src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -497,16 +497,12 @@ void IrcMessageHandler::handleUserStateMessage(Communi::IrcMessage *message)
     auto currentUser = getApp()->accounts->twitch.getCurrent();
 
     // set received emote-sets, used in TwitchAccount::loadUserstateEmotes
-    bool shouldLoadEmotes = currentUser->areLastUserstateEmoteSetsEmpty();
-
-    currentUser->setLastUserstateEmoteSets(
+    bool emoteSetsChanged = currentUser->setUserstateEmoteSets(
         message->tag("emote-sets").toString().split(","));
 
-    // we have to call loadUserstateEmotes if this is a first USERSTATE
-    // after currentUserChanged signal emitted
-    if (shouldLoadEmotes)
+    if (emoteSetsChanged)
     {
-        // TODO: figure out a nice way to reload only emotes from USERSTATE
+        currentUser->loadUserstateEmotes();
```

Comment:
Note for the future:
My biggest concern about calling `loadUserstateEmotes` here is that even though we filter out already fetched emotesets, we might append some emotes to `emoteData` without flushing it. It could lead to some emotesets being loaded in twice.
Knowing Twitch something like emoteset changing ID could happen. Leaving the comment just if something like this could ever occur.